run/authentication/auth.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2020 Google, LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START run_service_to_service_auth]
```

Comment:
Hrm, the word "function" means too many things.  Would the Cloud Functions sample be a "Function" with a capital F or would it just be a function?  If it's the former, is it an unauthenticated HTTP function? Or is it just a function to show how to access an authenticated Function?   

I'm confusing myself.  

My main concern is that if it's a Function, it takes a request as its input, which is different from taking a URL like in this sample.

Reply:
Because this sample is not actually a Cloud Function, but instead a snippet that could be used inside a Cloud Function or a Cloud Run service, this could be done as a sample common to both products.

An example of this is https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/functions/tips/main.py#L66, where two different region tags wrap the sample.

I'm suggesting that instead of adding a new sample, the following could be done:
* Use the code here to update https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/functions/security/main.py?rgh-link-date=2020-08-01T01%3A14%3A13Z#L15
* Add the region tag here as a second region tag wrapping the GCF sample
* Potentially move to a location where general auth samples are maintained

Reply:
I've moved it to the auth folder.   Would like to do a follow-up CL to update the functions code -- it's used [here](https://cloud.google.com/functions/docs/securing/authenticating).  I would need to update the tutorial and have the set-up for the cloud Function be separate from the auth call.